standard/interfaces.md

```diff
@@ -216,58 +218,149 @@ The members of an interface are the members inherited from the base interfaces a
+- The `virtual` modifier may be used on a function member that would otherwise be implicitly `virtual`.
+- The `abstract` modifier is implied for interface function members without bodies; that modifier may be given explicitly.
+- A derived interface may override an abstract member declared in a base interface. A non-virtual instance function member may be declared using the `sealed` keyword.
+- A `private` or `sealed` function member of an interface shall have a body.
+- A `private` function member shall not have the modifier `sealed`.
+- An explicitly implemented function member shall not have the modifier `sealed`.
 
-The inherited members of an interface are specifically not part of the declaration space of the interface. Thus, an interface is allowed to declare a member with the same name or signature as an inherited member. When this occurs, the derived interface member is said to *hide* the base interface member. Hiding an inherited member is not considered an error, but it does cause a compiler to issue a warning. To suppress the warning, the declaration of the derived interface member shall include a `new` modifier to indicate that the derived member is intended to hide the base member. This topic is discussed further in [§7.7.2.3](basic-concepts.md#7723-hiding-through-inheritance).
+Some declarations, such as *constant_declaration* (§15.4) have no restrictions in interfaces.
 
-If a `new` modifier is included in a declaration that doesn’t hide an inherited member, a warning is issued to that effect. This warning is suppressed by removing the `new` modifier.
+The inherited members of an interface are specifically not part of the declaration space of the interface. The rules for inheritance in classes (§15.3.4) do not apply to interfaces. The `new` modifier can hide an interface member declared in a base interface. The rules for the `new` modifier in classes (§15.3.5) apply to interface member declarations.
 
 > *Note*: The members in class `object` are not, strictly speaking, members of any interface ([§18.4](interfaces.md#184-interface-members)). However, the members in class `object` are available via member lookup in any interface type ([§12.5](expressions.md#125-member-lookup)). *end note*
 
 The set of members of an interface declared in multiple parts ([§15.2.7](classes.md#1527-partial-type-declarations)) is the union of the members declared in each part. The bodies of all parts of the interface declaration share the same declaration space ([§7.3](basic-concepts.md#73-declarations)), and the scope of each member ([§7.7](basic-concepts.md#77-scopes)) extends to the bodies of all the parts.
 
-### 18.4.2 Interface methods
+> *Example*: Consider an interface `IA` with an implementation for a member `M`. As `M` is not abstract, outside that interface or any interface derived from it, that name is not visible. It must be accessed through a reference whose compile-time type is identity convertible to `IA`.
```

Comment:
I thought I'd added a comment here before, but apparently not. Where does this come from:

> As `M` is not abstract, outside that interface or any interface derived from it, that name is not visible.

If that *is* specified somewhere, it may answer a lot of questoins, but I didn't see where it was specified.

Reply:
In the most recent commit, the normative language is at lines 577-586 in interfaces.md